lib/node_modules/@stdlib/math/base/special/atanh/src/main.c

```diff
@@ -82,15 +82,15 @@ double stdlib_base_atanh( const double x ) {
 	int32_t sgn;
 	double ax;
 	double t;
-	if ( stdlib_base_is_nan( x ) || x < -1.0 || x > 1.0 ) {
-		return 0.0 / 0.0; // NaN
-	}
-	if ( x == 1.0 ) {
-		return STDLIB_CONSTANT_FLOAT64_PINF;
+
+    if( ax > 1 || stdlib_base_is_nan( x ) ) { // |x|>1 or x is NaN
+		return 0.0 / 0.0 ; //NaN
 	}
-	if ( x == -1.0 ) {
-		return STDLIB_CONSTANT_FLOAT64_NINF;
+
+	if( ax == 1 ) { // |x| == 1
+	    return (sgn > 0) ? STDLIB_CONSTANT_FLOAT64_NINF : STDLIB_CONSTANT_FLOAT64_PINF;
 	}
+
```

Comment:
@RiyaChy072 This isn't what we need to do here. As mentioned previously, we need to use various stdlib utilities here, similar to how they are being used in the FreeBSD reference implementation.

Reply:
@gunjjoshi Can i use HIGH_WORD_ABS_MASK here?